### DISCUSSION TITLE
how to prevent tree-view left pane when opening a single file?

### DISCUSSION THREAD

**Discussion by ijl20** [atom/atom#22308] [Q&A]
I find I most often do not need the left-pane tree view open. Atom is my editor-of-choice so it's configured to launch when I inspect many file types by clicking in the OS (Linux, Windows) file explorer. So I find my most common file-open sequence has become:
1. click file to view/edit in Atom
2. close the empty left pane tree view.

After maybe 2 years of Atom (it's better than VS Code IMHO) I've discovered I can 'disable' the tree-view package and this gives me the default behaviour I need without breaking the editor. The ftp-remote-edit package (which I use a LOT and is a particular reason to like Atom over VS Code) displays its own tree view without using the tree-view package, so that is an unexpected major plus given my drastic hack to stop the tree-view pane appearing by default. E.g. "atom .bashrc" gives:

![atom_tree-view](https://user-images.githubusercontent.com/18136373/116525804-cbc0b480-a8d0-11eb-917d-02b52948b7b8.png)

I don't dislike the tree-view package, and disabling it seems way too drastic. 

**Is there a way to NOT have the tree-view pane appearing by default when I open a file.**

**Comment by PushkaCom** [comment]
Answer from
https://discuss.atom.io/t/hide-treeview-by-default/7718/44

**Into this file:**
~/.atom/init.coffee
**Add this code:**

```
atom.workspace.onDidOpen ({item}) ->
  itemName = item.constructor.name
  if (itemName  != 'TreeView') # no action if open TreeView intentionally
    dock = atom.workspace.paneContainerForURI('atom://tree-view')
    if dock && dock.isVisible()
      dock.hide()
```

**Reply by ThatXliner** [reply]
\:O that doesn't seem too hacky, either
